### DISCUSSION TITLE
Tuya Devices Protocol Question - UDP Multicast/Broadcast

### DISCUSSION THREAD

**Discussion by rlarranaga** [codetheweb/tuyapi#495] [General]
Hello,
i have a number of Tuya devices connected to my network, mostly light bulbs. i control them using the tuya-mqtt project on openHAB.
While capturing traffic on my network i noticed the devices send UDP Multicast.Broadcast packets every couple of seconds.

Does anyone know if there is a way to stop this behavior? Broadcast and multicast is usually transmitted in the lowest Wifi rates, so it is not great to have these devices broadcasting continuously. Blocking the traffic would not prevent the device from trying to send the broadcast and occupy the air resource either.

I read on General Discussion #5 the following:

"Connections & Broadcasts
When the device is not connected, it sends out UDP broadcast packets to the network every 3-6 seconds."

What does it mean that the broadcasts are sent when the device is not connected? I would have though the devices are connected to tuya-mqtt since i can control them. Any chance i can get some specifics on what kind of connection the devices needs to stop sending UDP packets?

as mentioned, my goal is mostly preventing the devices to send so many udp packets as opposed to block the packets.

Thanks!

**Comment by codetheweb** [comment]
I think the info in #5 may be outdated. AFAIK there's not a way to get your device to stop sending broadcasts besides flashing custom firmware (which may be a good option, depending on your device model).

**Reply by rlarranaga** [reply]
Hey @codetheweb:
Thanks for confirming.
If find this a bit frustrating from the point of view of tuya. I count 10 tuya devices in my network, all broadcasting every 3 seconds. Broadcasts use basic rates, so it is not great for resource utilization.

I have actually 14 devices. 4 of them have ESP8266 chips inside, so for those i was able to change the firmware. The others have a wb2l chip in them, so i haven't found any alternatives (If anyone knows of any, please let me know, i would definitely prefer an open firmware).

For now what i am trying to do is to disable the slowest rates, but that solution would not scale, and depends on your topology and noise environment. Do you know any place i could find out more information about the protocol? I might start poking these to see if i can make the broadcasts stop or slow down somehow

I have second question, that is unrelated. To connect the devices to tuya-mqtt i went the tuya-iot developer account way.
Does anyone knows what happens when the developer account trial expires?

Thanks for your help!

**Reply by codetheweb** [reply]
Documentation is kinda scattered. There's some useful info in [the wiki](https://github.com/codetheweb/tuyapi/wiki/Communication-Details). You might also find Tuya's [official documentation](https://developer.tuya.com/en/docs/iot) helpful.

You should be able to create a new cloud project when your trial expires.
